mkdd_widgets.py

```diff
@@ -1183,13 +1186,19 @@ def paintGL(self):
 
                         if start_point_selected or end_point_selected:
                             checkpoints_to_highlight.add(count)
+
+                        if is_sectionpoint:
+                            section_points.add(checkpoint)
+
                         count += 1
 
                     # Draw the lines between the points
                     glBegin(GL_LINES)
                     for checkpoint in group.points:
                         if checkpoint in concave_checkpoints:
                             glColor3f(1.0, 0.0, 0.0)
+                        elif checkpoint in section_points:
+                            glColor3f(*lap_checkpoint_color) # Mark lap/section checkpoints in a fancy green
```

Comment:
My preference would be to omit this inline comment. The code alone should be self-explanatory and, if the color in `lap_checkpoint_color` changed in the future (e.g. to purple), then we'd need to remember to update this inline comment.